azure-pipelines.yml

```diff
@@ -11,26 +11,72 @@ pr:
       - develop
   drafts: false
 
-pool:
-  vmImage: ubuntu-latest
+variables:
+  environmentName: 'team-up-dev'
+  vmImageName: 'ubuntu-latest'
 
-steps:
-  - script: npm ci
-    displayName: 'Install Dependencies'
+stages:
+- stage: Build
+  displayName: Build stage
+  jobs:
+  - job: Build
+    displayName: Building
+    pool:
+      vmImage: $(vmImageName)
 
-  - script: npm run lint:fix
-    displayName: 'Linting'
+    steps:
+      - script: npm ci
+        displayName: 'Installing Dependencies'
 
-  - script: npm test -- --ci --watchAll=false --passWithNoTests
-    displayName: 'Running Unit Test'
+      - script: npm run lint:fix
+        displayName: 'Linting'
 
-  - script: npm run build
-    displayName: 'Build React webApp'
+      - script: npm test -- --ci --watchAll=false --passWithNoTests
+        displayName: 'Running Unit Test'
 
-  - task: PublishBuildArtifacts@1
-    displayName: 'Save Artifact'
-    condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-    inputs:
-      PathtoPublish: '$(Agent.BuildDirectory)/s/build'
-      ArtifactName: 'drop'
-      publishLocation: 'Container'
+      - script: npm run build-prod
```

Comment:
if you're going to use this, you're repeating tasks behind the scenes because build-prod===npm ci && npm run format && npm run build from package json, right?

Reply:
yep, sorry, fixing

Reply:
hey, let's change this for npm run build, because we're gathering dependencies twice